code/Fluent.php

```diff
@@ -669,4 +669,40 @@ public static function get_template_global_variables()
             )
         );
     }
+
+    /**
+     * Given a field on an object and optionally a locale, compare its locale value against the default locale value to
+     * determine if the value is changed at the given locale.
+     *
+     * @param  DataObject  $object
+     * @param  FormField   $field
+     * @param  string|null $locale Optional: if not provided, will be gathered from the request
+     * @return boolean
+     */
+    public static function isFieldModified(DataObject $object, FormField $field, $locale = null)
+    {
+        if (is_null($locale)) {
+            $locale = self::current_locale();
+        }
+
+        if ($locale === $defaultLocale = self::default_locale()) {
+            // It's the default locale, so it's never "modified" from the default locale value
+            return false;
+        }
+
+        $defaultField = self::db_field_for_locale($field->getName(), $defaultLocale);
+        $localeField  = self::db_field_for_locale($field->getName(), $locale);
+
+        $defaultValue = $object->$defaultField;
+        $localeValue  = $object->$localeField;
+
+        if ((!is_null($defaultValue) && is_null($localeValue))
```

Comment:
Actually, string values can be "mapped" to null, so can we use `empty()` instead of `is_null` to check for has-value?

Fluent fields do not distinguish between null / empty, since that distinction is often ambiguous in core. Thus we are not able to use, for example, `null` for untranslated, or '' for translated as empty.

The problem is that you can end up with empty strings appearing as though they are modified, but they aren't really. :P